### PR TITLE
Add context handling to go-routesapi methods

### PR DESCRIPTION
**Pull request by daniel-de-vera**

### CODE REVIEW

File: go-routesapi/watched/baseline_watched.go [hunk 1]
```diff
@@ -15,10 +15,16 @@ type BaselineWatched interface {
 	// the routing key rk for the associated baseline workload.
 	Get(rk string) *routesapi.WorkloadRoutingRule
 
+	// Same as Get() but with context handling
+	GetContext(ctx context.Context, rk string) (*routesapi.WorkloadRoutingRule, error)
```
Comment (scott-cotton) [inline go-routesapi/watched/baseline_watched.go:19]: I think I would prefer these be named with Context as prefix as opposed to suffix, or with a `WithContext` suffix to help avoid reading it as something that gets context (and likewise the other methods).

**Review by scott-cotton — APPROVED** [review]
LGTM,
left a couple of non-blocking stylistic comments

File: go-routesapi/watched/baseline_watched.go (hunk 1 above)
Comment (daniel-de-vera) [reply go-routesapi/watched/baseline_watched.go:19]: I was mimicking https://pkg.go.dev/net/http#Transport -> `Dial/DialContext`, `DialTLS/DialTLSContext`, but sure I can rename then.
Which of the two option do you prefer?

Comment (scott-cotton) [reply go-routesapi/watched/baseline_watched.go:19]: Yeah I see there are other functions with suffix Context, it's just hard to read `GetContext` and grok what it means.  I guess I'd prefer `WithContext` when the compound name tends to mean something else, package normalised.

Comment (daniel-de-vera) [reply go-routesapi/watched/baseline_watched.go]: Renamed in https://github.com/signadot/routesapi/pull/13/commits/18eed71c326c728e0cd8a5402fb56f4409446fc6.

**Pull request merged**